Makefile

```diff
@@ -31,6 +31,10 @@ include $(DMLC_CORE)/make/dmlc.mk
 WARNFLAGS= -Wall
 CFLAGS = -DMSHADOW_FORCE_STREAM $(WARNFLAGS)
 
+ifeq ($(DEV), 1)
+	CFLAGS += -g -Werror
```

Comment:
you don't need -g just to generate symbol. I think there is an option to do it without turning off optimization